pubsub/cli.rb

```diff
@@ -0,0 +1,127 @@
+#!/usr/bin/env ruby
+# Copyright 2015 Google, Inc
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require "gcloud"
+
+# Shows using gcloud to interact with Pub/Sub
+class PubSubSample
+  attr_accessor :topic, :sub
+
+  def initialize project
+    @project = project
+    @topic = "ruby-sample"
+    @sub = "ruby-sample-sub"
+    @subp = "ruby-sample-sub-p"
+
+    gcloud = Gcloud.new @project
+    pubsub = gcloud.pubsub
+
+    topic = pubsub.topic @topic, autocreate: true
+
+    sub = topic.get_subscription @sub
+    topic.subscribe @sub if sub.nil?
```

Comment:
I thought 1 line blocks were bad, so:
```ruby
subscription = topic.subscription subscription_name
subscription = topic.subscribe subscription_name unless subscription.exists?
```
in this case we don't need the second `subscription =` because the variable is not used.